### PR TITLE
vala 0.52.2

### PR DESCRIPTION
**Pull request by Gottox**
- vala: update to 0.52.2.
- libvaladoc: update to 0.52.2.

### CODE REVIEW

File: srcpkgs/vala/template [hunk 1]
```diff
@@ -14,16 +14,17 @@ license="LGPL-2.1-or-later"
...
 
 post_install() {
 	# remove the /usr/bin prefix from the vapigen variable
 	# so the package that use it fall into our wrapper.
 	# keep the binary versioned, we don't know the potential
 	# fallout from using the unversioned one
-	vsed -e 's|^vapigen=.*|vapigen=vapigen-0.50|g' \
-		 -i ${DESTDIR}/usr/lib/pkgconfig/vapigen-0.50.pc
+	vsed -e 's|^vapigen=.*|vapigen=vapigen-0.52|g' \
+		 -i ${DESTDIR}/usr/lib/pkgconfig/vapigen-0.52.pc
```
Comment (ericonr) [inline srcpkgs/vala/template:27]: Use `${version%.*}` ?

**Review by ericonr — COMMENTED** [review]
`libvaladoc` commit should be `valadoc: ...` and it's missing changes to template.

**Comment by Gottox** [comment]
vala tests only break for the ci.

**Comment by paper42** [comment]
The vala patch now doesn't apply because patch_args="-Np1" is the new default

**Comment by ericonr** [comment]
@paper42 I believe Gottox mentioned not being entirely available for a while. I think it should be fine to open a PR to supersede this one, if there's anything still missing.

**Comment by ericonr** [comment]
Merged in #32351

**Pull request closed without merging**